docs/cow-protocol/tutorials/solvers/flashloans_support.md

```diff
@@ -0,0 +1,77 @@
+
+As you can see the `flashloan` field contains just a few pieces of information:
+- `amount`: units (in wei) that should be flashloaned
+- `token`: which token should be flashloaned
+- `receiver`: address that is supposed to receive the tokens
+- `liquidityProvider`: the lending protocol’s contract (i.e. AavePool)
+- `protocolAdapter`: which address needs to be used to adapt the `liquidityProvider` to CoW Protocol’s flashloan machinery.
+
+The reference driver is already doing all that for you and is also propagating the flashloan data to the connected solver endpoint as a field on the `order` struct. For example, [this](https://solver-instances.s3.eu-central-1.amazonaws.com/staging/xdai/auction/146823591.json) is a `/solve` request that your solver will get from the driver (just search for `flashloan` and you will find it).
+
+## Adjustments needed in the solver/driver
+
+### Encode the Call to the Flashloan Router
+
+Because the flashloan needs to be available over the entire duration of the settlement, all flashloan settlements will have to go through the [flashloan router](https://docs.cow.fi/cow-protocol/reference/contracts/periphery/flashloans#iflashloanrouter-contract) (deterministically deployed at `0x9da8b48441583a2b93e2ef8213aad0ec0b392c69`). So instead of calling `settle()` on the settlement contract you have to call `flashloanAndSettle()` on the flashloan router contract. This is straightforward, and only takes 2 arguments:
+1. `Loan.Data[]`, which can be trivially initialized with the data from the order’s `flashloan` field;
+2. the settlement, which is the calldata for the regular `settle()` call the router will eventually initiate after taking out all the flashloans.
+
+:::note
+
+Specifically for the AAVE integration, the driver is not required to inject any additional permissioned interactions. If you call the flashloan router correctly and include the pre- and post-hooks of the order the call should pass.
+
+This is completely handled by the reference driver.
+
+:::
+
+### Estimating the Gas Costs
+
+Your solver should know how much gas overhead the flashloan logic adds to the solution in order to capture enough network fees to cover the total gas cost. If this overhead is not taken into account, every time your solver settles a flashloan order would result in losses due to the underestimated gas costs. Proper gas estimation will be needed for all solvers - even the ones connected to the reference driver as the reference driver, as the reference driver does not do any post-processing and adjustments of the network fees computed by solvers.
```

Comment:
I think nicer introduction start something like. 

Solvers are responsible of estimating how much gas it takes to settle a batch. Flashloan flow requires a call to a wrapper contract, which takes the loans and then call to the settlement contract. This wrapper obviously adds an overhead compared to just calling the settlement contract, for this reason solvers should estimate the total gas needed.